Terminal.Gui/View/Layout/Dim.cs

```diff
@@ -0,0 +1,766 @@
+
+        // Factor in adornments
+        Thickness thickness = us.GetAdornmentsThickness ();
+
+        if (dimension == Dimension.Width)
+        {
+            max += thickness.Horizontal;
+        }
+        else
+        {
+            max += thickness.Vertical;
+        }
+
+        // If max: is set, clamp the return - BUGBUG: Not tested
+        return int.Min (max, MaximumContentDim?.Anchor (superviewContentSize) ?? max);
+    }
+
+    internal override bool ReferencesOtherViews ()
+    {
+        // BUGBUG: This is not correct. _contentSize may be null.
+        return false; //_style.HasFlag (DimAutoStyle.Content);
+    }
+}
+
+/// <summary>
+///     Represents a dimension that is a combination of two other dimensions.
+/// </summary>
+/// <param name="add">
+///     Indicates whether the two dimensions are added or subtracted. If <see langword="true"/>, the dimensions are added,
+///     otherwise they are subtracted.
+/// </param>
+/// <remarks>
+///     This is a low-level API that is typically used internally by the layout system. Use the various static
+///     methods on the <see cref="Dim"/> class to create <see cref="Dim"/> objects instead.
+/// </remarks>
+/// <param name="left">The left dimension.</param>
+/// <param name="right">The right dimension.</param>
+public class DimCombine (bool add, Dim left, Dim right) : Dim
+{
+    /// <summary>
+    ///     Gets whether the two dimensions are added or subtracted.
+    /// </summary>
+    public bool Add { get; } = add;
+
+    /// <summary>
+    ///     Gets the left dimension.
+    /// </summary>
+    public Dim Left { get; } = left;
+
+    /// <summary>
+    ///     Gets the right dimension.
+    /// </summary>
+    public Dim Right { get; } = right;
+
+    /// <inheritdoc/>
+    public override string ToString () { return $"Combine({Left}{(Add ? '+' : '-')}{Right})"; }
+
+    internal override int Anchor (int size)
+    {
+        int la = Left.Anchor (size);
+        int ra = Right.Anchor (size);
+
+        if (Add)
+        {
+            return la + ra;
+        }
+
+        return la - ra;
+    }
+
+    internal override int Calculate (int location, int superviewContentSize, View us, Dimension dimension)
+    {
+        int leftNewDim = Left.Calculate (location, superviewContentSize, us, dimension);
+        int rightNewDim = Right.Calculate (location, superviewContentSize, us, dimension);
+
+        int newDimension;
+
+        if (Add)
+        {
+            newDimension = leftNewDim + rightNewDim;
+        }
+        else
+        {
+            newDimension = Math.Max (0, leftNewDim - rightNewDim);
+        }
+
+        return newDimension;
+    }
+
+    /// <summary>
+    ///     Diagnostics API to determine if this Dim object references other views.
+    /// </summary>
+    /// <returns></returns>
+    internal override bool ReferencesOtherViews ()
+    {
+        if (Left.ReferencesOtherViews ())
+        {
+            return true;
```

Comment:
Are these possible to reach? 🤔 

Not tracing them manually, it looks like only false is possible, right here, anyway, because the recursive default is false, making it false up the stack.

Also potentially a really good place for some ref love, because the compiler can massively optimize this sort of case at runtime especially in later tiers, but that needs careful implementation, so just a note.